bin/compile

```diff
@@ -24,10 +15,26 @@ if [ -f "$ENV_DIR/CUSTOM_ARTIFACT_URL" ]; then
    CUSTOM_ARTIFACT_URL=`cat $ENV_DIR/CUSTOM_ARTIFACT_URL`
 fi
 
+# file name, eg. event.processor.tar.gz
+ARTIFACT=`cat $ENV_DIR/ARTIFACT`
+# build id, eg 1.0-2017.03.27-82178560e00eaa688807cd5f04faef472d0f2db9-1234
+BUILD_ID=`cat $ENV_DIR/BUILD_ID`
+
 if [ -n "$CUSTOM_ARTIFACT_URL" ]; then
     wget "$CUSTOM_ARTIFACT_URL" -O $LOCAL_ARTIFACT
 else
-    wget "https://${BUILD_NUMBER}-${CIRCLE_REPO_NUM}-gh.circle-artifacts.com/0${ARTIFACT}?circle-token=${CIRCLE_TOKEN}" -O $LOCAL_ARTIFACT
+    # grab the artifact from s3
+    AMZ_FILE="$BUILD_ID/$ARTIFACT"
+    BUCKET="ddp-build"
+    AWS_KEY=`cat $ENV_DIR/AWS_ACCESS_KEY_ID`
+    AWS_SECRET=`cat $ENV_DIR/AWS_SECRET_ACCESS_KEY`
+    mkdir ~/.aws
+    cat > ~/.aws/credentials <<EOF
+[default]
+aws_access_key_id = ${AWS_KEY}
+aws_secret_access_key = ${AWS_SECRET}
+EOF
```

Comment:
Let's try this with env vars only. Also, specify AWS_DEFAULT_REGION as us-west-2.